tests/jax/test_fused_attn.py

```diff
@@ -338,42 +387,53 @@ def check_dqkv(primitive, reference, valid_len):
     pytest.param(AttnMaskType.PADDING_CAUSAL_MASK, id='PADDING_CAUSAL'),
 ])
 @pytest.mark.parametrize('qkv_layout', [
-    pytest.param(QKVLayout.BS3HD, id='qkvpacked'),
-    pytest.param(QKVLayout.BSHD_BS2HD, id='kvpacked'),
-    pytest.param(QKVLayout.BSHD_BSHD_BSHD, id='separate'),
+    pytest.param(QKVLayout.BS3HD, id='QKV_PACKED'),
+    pytest.param(QKVLayout.BSHD_BS2HD, id='KV_PACKED'),
+    pytest.param(QKVLayout.BSHD_BSHD_BSHD, id='SEPARATE'),
+])
+@pytest.mark.parametrize('dtype', [
+    pytest.param(jnp.bfloat16, id="BF16"),
+    pytest.param(jnp.float16, id="FP16")
+])
+@pytest.mark.parametrize('b, s_q, s_kv, h_q, h_kv, d',[
+    pytest.param(32,  128,  128, 16, 16, 64, id='SELF_512'),
+    pytest.param( 4, 2048, 2048, 12, 12, 64, id='SELF_ARB'),
+    pytest.param(32,  512,  128, 16, 16, 64, id='CROSS_512'),
+    pytest.param( 4, 2048, 1024, 12, 12, 64, id='CROSS_ARB'),
+    pytest.param(32,  128,  128, 12,  6, 64, id='GQA_512'),
+    pytest.param( 4, 2048, 2048, 12,  6, 64, id='GQA_ARB')
 ])
-@pytest.mark.parametrize('dropout_prob', [0., 0.1])
-@pytest.mark.parametrize('is_training',
-                         [pytest.param(True, id='training'),
-                          pytest.param(False, id='inference')])
-@pytest.mark.parametrize(
-    'dtype', [pytest.param(jnp.bfloat16, id="BF16"),
-              pytest.param(jnp.float16, id="FP16")])
-@pytest.mark.parametrize('b, s_q, s_kv, h_q, h_kv, d',
-                         [(32, 128, 128, 16, 16, 64), (4, 2048, 2048, 12, 12, 64),
-                          pytest.param(32, 512, 128, 16, 16, 64, id='32-512-128-16-16-64-cross'),
-                          pytest.param(4, 2048, 2048, 12, 6, 64, id='4-2048-2048-12-6-64-GQA')])
 class TestFusedAttn:
     """
     Fused attention tester
     """
 
+    @pytest.mark.parametrize('dropout_prob', [
+        pytest.param(0.0, id="DROP_0.0"),
+        pytest.param(0.1, id="DROP_0.1")
+    ])
+    @pytest.mark.parametrize('is_training', [
+        pytest.param(True, id='TRAINING'),
+        pytest.param(False, id='INFERENCE'),
+    ])
     @staticmethod
-    def test_forward(b, s_q, s_kv, h_q, h_kv, d, attn_bias_type, attn_mask_type, dropout_prob,
-                     dtype, is_training, qkv_layout):
+    def test_forward(b, s_q, s_kv, h_q, h_kv, d, attn_bias_type, attn_mask_type,
+                     dropout_prob, dtype, is_training, qkv_layout, bias_shape):
         """
         Test forward with parameterized configs
         """
+        pytest.skipif(is_training and dropout_prob > 0.,
```

Comment:
oh, I just see you move the skipif to here, but this is not the original intend. The original tests still run the custom call for the sanity check and only skips the value checks. However, moving here then we will lose the sanity check for the dropout.

Reply:
Ah, that makes sense. I missed this and thought that we were basically just doing a lot of wasteful compute here without testing anything. I'll move it back to get the sanity check.